test/unit/test_cluster_fuzz.py

```diff
@@ -0,0 +1,248 @@
+import os
+import re
+import statistics
+import subprocess
+import sys
+import tarfile
+import tempfile
+
+from scripts.test import shared
+from . import utils
+
+
+class ClusterFuzz(utils.BinaryenTestCase):
+    @classmethod
+    def setUpClass(cls):
+        # Bundle up our ClusterFuzz package, and unbundle it to a directory.
+        # Keep the directory alive in a class var.
+        cls.temp_dir = tempfile.TemporaryDirectory()
+        cls.clusterfuzz_dir = cls.temp_dir.name
+
+        bundle = os.environ.get('BINARYEN_CLUSTER_FUZZ_BUNDLE')
+        if bundle:
+            print(f'Using existing bundle: {bundle}')
+        else:
+            print('Making a new bundle')
+            bundle = os.path.join(cls.clusterfuzz_dir, 'bundle.tgz')
+            cmd = [shared.in_binaryen('scripts', 'bundle_clusterfuzz.py')]
+            cmd.append(bundle)
+            # wasm-opt is in the bin/ dir, and the build dir is one above it,
+            # and contains bin/ and lib/.
+            build_dir = os.path.dirname(os.path.dirname(shared.WASM_OPT[0]))
+            cmd.append(f'--build-dir={build_dir}')
+            shared.run_process(cmd)
+
+        print('Unpacking bundle')
+        tar = tarfile.open(bundle, "r:gz")
+        tar.extractall(path=cls.clusterfuzz_dir)
+        tar.close()
+
+        print('Ready')
+
+    # Test our bundler for ClusterFuzz.
+    def test_bundle(self):
+        # The bundle should contain certain files:
+        # 1. run.py, the main entry point.
+        assert os.path.exists(os.path.join(self.clusterfuzz_dir, 'run.py'))
```

Comment:
We should use the python unit test assertion methods rather than raw `assert` if this is going to be run as a unit test.

Reply:
Good point, done.